src/ripple/shamap/SHAMapTreeNode.h

```diff
@@ -301,35 +301,21 @@ class SHAMapInnerNode : public SHAMapAbstractNode,
 
 // SHAMapTreeNode represents a leaf, and may eventually be renamed to reflect
 // that.
-class SHAMapTreeNode : public SHAMapAbstractNode,
-                       public CountedObject<SHAMapTreeNode>
+class SHAMapTreeNode : public SHAMapAbstractNode
 {
-private:
+protected:
     std::shared_ptr<SHAMapItem const> mItem;
-    SHAMapNodeType mType;
 
 public:
     SHAMapTreeNode(const SHAMapTreeNode&) = delete;
     SHAMapTreeNode&
     operator=(const SHAMapTreeNode&) = delete;
 
+    SHAMapTreeNode(std::shared_ptr<SHAMapItem const> item, std::uint32_t seq);
     SHAMapTreeNode(
         std::shared_ptr<SHAMapItem const> item,
-        SHAMapNodeType type,
-        std::uint32_t seq);
-    SHAMapTreeNode(
-        std::shared_ptr<SHAMapItem const> item,
-        SHAMapNodeType type,
         std::uint32_t seq,
         SHAMapHash const& hash);
-    std::shared_ptr<SHAMapAbstractNode>
-    clone(std::uint32_t seq) const override;
-
-    SHAMapNodeType
-    getType() const override
-    {
-        return mType;
-    }
 
     bool
     isLeaf() const override
```

Comment:
Reopening this so this comment is seen. Marking these function `final` without the `override` is causing gcc8 to warn; we need both the `final` and the `override`. However, some of these classes are already marked as `final`. For `final` classes, I would not mark these functions as `final` (just to cut down on the noise of `final override`). I'll make a patch to resolve the warnings.

Reply:
@nbougalis 
Here's a patch to reformat with clang-format: https://github.com/seelabs/rippled/commit/e9c5db58921b5292bd3571fd9ee8965c53dc7e3f
Here's a patch to resolve the `suggest-override` warnings: https://github.com/seelabs/rippled/commit/814a52959fc8f409d6578c1def827884487807c9

Hopefully this will resolve the CI issues. Also note, I'm not able to test those locally, as I do not have access to gcc 8.